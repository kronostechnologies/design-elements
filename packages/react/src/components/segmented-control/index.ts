export { SegmentedControl } from './segmented-control';
export type { SegmentedControlProps } from './segmented-control';
