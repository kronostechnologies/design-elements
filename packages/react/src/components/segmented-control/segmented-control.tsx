import { MouseEvent, useState, useEffect, VoidFunctionComponent } from 'react';
import styled from 'styled-components';
import { focus } from '../../utils/css-state';
import { Icon, IconName } from '../icon/icon';
import { useDeviceContext } from '../device-context-provider/device-context-provider';

const Container = styled.div`
    background-color: ${({ theme }) => (theme.component['segmented-control-list-background-color'])};
    border-radius: var(--border-radius);
    display: flex;
    width: fit-content;
`;

interface ToggleButtonProps {
    pressed: boolean;
    isMobile: boolean;
    ariaDisabled: boolean;
}

const ToggleButton = styled.button<ToggleButtonProps>`
    align-items: center;
    background-color: ${({ theme, pressed }) => (pressed ? theme.component['segmented-control-pressed-background-color'] : theme.component['segmented-control-background-color'])};
    border: 1px solid ${({ theme, pressed }) => (pressed ? theme.component['segmented-control-pressed-border-color'] : theme.component['segmented-control-border-color'])};
    border-radius: var(--border-radius);
    color: ${({ theme, pressed }) => (pressed ? theme.component['segmented-control-pressed-text-color'] : theme.component['segmented-control-text-color'])};
    display: flex;
    font-size: ${({ isMobile }) => (isMobile ? '1rem' : '0.875rem')};
    font-weight: ${({ pressed }) => (pressed ? 'var(--font-semi-bold)' : 'var(--font-normal)')};
    gap: var(--spacing-1x);
    justify-content: center;
    letter-spacing: 0.05rem;
    line-height: 1.25rem;
    min-height: ${({ isMobile }) => (isMobile ? 'var(--size-3x)' : 'var(--size-2x)')};
    padding: 0 var(--spacing-1halfx);
    position: relative;

    &:hover {
        background-color: ${({ theme, pressed }) => (pressed ? theme.component['segmented-control-pressed-hover-background-color'] : 'transparent')};
        color: ${({ theme, pressed }) => (pressed ? theme.component['segmented-control-pressed-hover-text-color'] : theme.component['segmented-control-hover-text-color'])};

        &::before {
            background-color: ${({ theme }) => (theme.component['segmented-control-hover-background-color'])};
            border-radius: var(--border-radius-1halfx);
            content: ${({ pressed }) => (pressed ? 'none' : '" "')};
            display: block;
            left: 6px;
            min-height: 1.5rem;
            position: absolute;
            top: .1875rem;
            width: calc(100% - var(--spacing-1halfx));
        }
    }

    &:focus {
        z-index: 1;
    }

    ${focus};

    &[aria-disabled="true"],
    &[aria-disabled="true"]:hover {
        color: ${({ theme }) => theme.component['segmented-control-disabled-text-color']};
        pointer-events: none;
    }

    // Prevent layout shift when the label changes from normal to bold upon selection.
    span::after {
        content: attr(data-text);
        display: block;
        font-weight: var(--font-semi-bold);
        height: 0;
        overflow: hidden;
        pointer-events: none;
        user-select: none;
        visibility: hidden;
    }
`;

interface SegmentedControlProps {
    /**
     * Takes an array of objects containing all the buttons needed
     */
    buttonGroup: {
        defaultPressed?: boolean;
        disabled?: boolean;
        icon?: IconName;
        ariaLabel?: string;
        ariaDisabled?: boolean;
        label?: string;
        value: string
    }[];
    className?: string;
    /**
     * Sets common name for all buttons
     */
    groupName: string;
    requireSelection?: boolean;
    onClick?(event: MouseEvent<HTMLButtonElement>): void;
}

export const SegmentedControl: VoidFunctionComponent<SegmentedControlProps> = ({
    buttonGroup,
    className,
    groupName,
    requireSelection = false,
    onClick,
}) => {
    const { isMobile } = useDeviceContext();
    const defaultPressedButton = buttonGroup.find((button) => button.defaultPressed);
    const [selectedButton, setSelectedButton] = useState<string>(
        defaultPressedButton ? defaultPressedButton.value : ''
    );

    useEffect(() => {
        if (requireSelection && !selectedButton && buttonGroup.length > 0) {
            setSelectedButton(buttonGroup[0].value);
        }
    }, [buttonGroup, selectedButton, requireSelection]);

    const handleClick = (event: MouseEvent<HTMLButtonElement>): void => {
        const button = buttonGroup.find((btn) => btn.value === event.currentTarget.value);
        const newValue = event.currentTarget.value;

        if (button?.ariaDisabled) {
            return;
        }

        if (requireSelection) {
            if (selectedButton !== newValue) {
                setSelectedButton(newValue);
            }
        } else {
            setSelectedButton(newValue === selectedButton ? '' : newValue);
        }

        onClick?.(event);
    };

    return (
        <Container className={className} role="group" aria-label={groupName}>
            {buttonGroup.map((button, i) => (
                <ToggleButton
                    aria-label={button.ariaLabel || undefined}
                    aria-disabled={button.ariaDisabled ? true : undefined}
                    aria-pressed={button.value === selectedButton}
                    pressed={button.value === selectedButton}
                    data-testid={`test-toggle-button-${i}`}
                    isMobile={isMobile}
                    key={`${groupName}-${button.value}`}
                    onClick={handleClick}
                    type="button"
                    value={button.value}
                    ariaDisabled={!!button.ariaDisabled}
                >
                    {button.icon && (
                        <Icon
                            aria-hidden="true"
                            name={button.icon}
                            size="16"
                        />
                    )}
                    {button.label && (
                        <span data-text={button.label}>
                            {button.label}
                        </span>
                    )}
                </ToggleButton>
            ))}
        </Container>
    );
};
