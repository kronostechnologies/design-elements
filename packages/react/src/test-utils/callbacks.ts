export function doNothing(): void {
    // do nothing
}
