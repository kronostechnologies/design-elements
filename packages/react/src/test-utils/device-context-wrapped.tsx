import React, { ReactElement } from 'react';
import { DeviceContextProvider, DeviceType } from '../components/device-context-provider/device-context-provider';

export function DeviceContextWrapped(children: ReactElement, device?: DeviceType): ReactElement {
    return (
        <DeviceContextProvider staticDevice={device}>
            {children}
        </DeviceContextProvider>
    );
}
