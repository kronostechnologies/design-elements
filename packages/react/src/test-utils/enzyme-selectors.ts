import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
