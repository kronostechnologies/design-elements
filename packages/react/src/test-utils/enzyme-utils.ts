import { ReactWrapper } from 'enzyme';
import { act } from 'react-dom/test-utils';

export async function waitForComponentToPaint<P = {}>(
    wrapper: ReactWrapper<P>,
    amount = 0,
): Promise<void> {
    await act(async () => {
        await new Promise((resolve) => setTimeout(resolve, amount));
        wrapper.update();
    });
}

export function expectFocusToBeOn(element: ReactWrapper): void {
    expect(document.activeElement).toBe(element.getDOMNode());
}
