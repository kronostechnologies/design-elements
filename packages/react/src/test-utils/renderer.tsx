import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
