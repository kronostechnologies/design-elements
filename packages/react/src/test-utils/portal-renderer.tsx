import { render, RenderResult } from '@testing-library/react';
import React, { ReactElement } from 'react';
import { DeviceType } from '../components/device-context-provider/device-context-provider';
import { AllProviders } from './renderer';

export function renderPortalWithProviders(
    component: ReactElement,
    device?: DeviceType,
): RenderResult {
    return render(<AllProviders device={device}>{component}</AllProviders>);
}
