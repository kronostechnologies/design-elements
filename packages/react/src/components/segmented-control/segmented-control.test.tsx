import { shallow } from 'enzyme';
import { getByTestId } from '../../test-utils/enzyme-selectors';
import { renderWithProviders } from '../../test-utils/renderer';
import { SegmentedControl } from './segmented-control';

const buttonGroup = [
    { label: 'Option 1', value: 'option1' },
    { label: 'Option 2', value: 'option2', defaultPressed: true },
    { label: 'Option 3', value: 'option3', disabled: true },
    { label: 'Option 4', value: 'option4' },
];

describe('SegmentedControl', () => {
    test('onClick callback is called when clicked', () => {
        const callback = jest.fn();
        const wrapper = shallow(<SegmentedControl onClick={callback} buttonGroup={buttonGroup} groupName="Test1" />);

        getByTestId(wrapper, 'test-toggle-button-2').simulate('click', { currentTarget: { value: 'test' } });
        expect(callback).toHaveBeenCalled();
    });

    test('Is default pressed', () => {
        const wrapper = shallow(<SegmentedControl buttonGroup={buttonGroup} groupName="Test2" />);

        expect(getByTestId(wrapper, 'test-toggle-button-1').prop('pressed')).toBe(true);
    });

    test('should have aria-pressed="true" for the default pressed button', () => {
        const wrapper = shallow(<SegmentedControl buttonGroup={buttonGroup} groupName="Test2" />);
        const button = getByTestId(wrapper, 'test-toggle-button-1');

        expect(button.prop('aria-pressed')).toBe(true);
    });

    test('should have aria-disabled="true" for disabled button', () => {
        const wrapper = shallow(<SegmentedControl buttonGroup={buttonGroup} groupName="Test" />);
        const button = getByTestId(wrapper, 'test-toggle-button-2');

        expect(button.prop('aria-disabled')).toBe(true);
    });

    test('Matches snapshot (desktop)', () => {
        const { container } = renderWithProviders(
            <SegmentedControl buttonGroup={buttonGroup} groupName="Test4" />,
            'desktop',
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    test('Matches snapshot (mobile)', () => {
        const { container } = renderWithProviders(
            <SegmentedControl buttonGroup={buttonGroup} groupName="Test4" />,
            'mobile',
        );

        expect(container.firstChild).toMatchSnapshot();
    });
});
