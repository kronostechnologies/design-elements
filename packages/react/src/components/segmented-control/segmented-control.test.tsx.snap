// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl Matches snapshot (desktop) 1`] = `
.c0 {
  background-color: rgb(0 0 0 / 0.05);
  border-radius: var(--border-radius);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: var(--border-radius);
  color: #60666E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: var(--font-family);
  font-size: 0.875rem;
  font-weight: var(--font-normal);
  gap: var(--spacing-1x);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.25rem;
  min-height: var(--size-2x);
  padding: 0 var(--spacing-1halfx);
  position: relative;
}

.c1:hover {
  background-color: transparent;
  color: #000000;
}

.c1:hover::before {
  background-color: rgb(0 0 0 / 0.05);
  border-radius: var(--border-radius);
  content: " ";
  display: block;
  height: calc(100% - 0.375rem);
  left: 0.1875rem;
  min-height: 1.5rem;
  position: absolute;
  top: 0.1875rem;
  width: calc(100% - 0.375rem);
}

.c1:focus {
  z-index: 1;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1[aria-disabled='true'],
.c1[aria-disabled='true']:hover {
  color: #B7BBC2;
  pointer-events: none;
}

.c1 span::after {
  content: attr(data-text);
  display: block;
  font-weight: var(--font-semi-bold);
  height: 0;
  overflow: hidden;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  visibility: hidden;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #E0F0F9;
  border: 1px solid #006296;
  border-radius: var(--border-radius);
  color: #003A5A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: var(--font-family);
  font-size: 0.875rem;
  font-weight: var(--font-semi-bold);
  gap: var(--spacing-1x);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.25rem;
  min-height: var(--size-2x);
  padding: 0 var(--spacing-1halfx);
  position: relative;
}

.c2:hover {
  background-color: #C2E2F4;
  color: #003A5A;
}

.c2:hover::before {
  background-color: rgb(0 0 0 / 0.05);
  border-radius: var(--border-radius);
  content: none;
  display: block;
  height: calc(100% - 0.375rem);
  left: 0.1875rem;
  min-height: 1.5rem;
  position: absolute;
  top: 0.1875rem;
  width: calc(100% - 0.375rem);
}

.c2:focus {
  z-index: 1;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2[aria-disabled='true'],
.c2[aria-disabled='true']:hover {
  color: #B7BBC2;
  pointer-events: none;
}

.c2 span::after {
  content: attr(data-text);
  display: block;
  font-weight: var(--font-semi-bold);
  height: 0;
  overflow: hidden;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  visibility: hidden;
}

<div
  aria-label="Test4"
  class="c0"
  role="group"
>
  <button
    aria-pressed="false"
    class="c1"
    data-testid="test-toggle-button-0"
    type="button"
    value="option1"
  >
    <span
      data-text="Option 1"
    >
      Option 1
    </span>
  </button>
  <button
    aria-pressed="true"
    class="c2"
    data-testid="test-toggle-button-1"
    type="button"
    value="option2"
  >
    <span
      data-text="Option 2"
    >
      Option 2
    </span>
  </button>
  <button
    aria-disabled="true"
    aria-pressed="false"
    class="c1"
    data-testid="test-toggle-button-2"
    type="button"
    value="option3"
  >
    <span
      data-text="Option 3"
    >
      Option 3
    </span>
  </button>
  <button
    aria-pressed="false"
    class="c1"
    data-testid="test-toggle-button-3"
    type="button"
    value="option4"
  >
    <span
      data-text="Option 4"
    >
      Option 4
    </span>
  </button>
</div>
`;

exports[`SegmentedControl Matches snapshot (mobile) 1`] = `
.c0 {
  background-color: rgb(0 0 0 / 0.05);
  border-radius: var(--border-radius);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: 1px solid transparent;
  border-radius: var(--border-radius);
  color: #60666E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: var(--font-family);
  font-size: 1rem;
  font-weight: var(--font-normal);
  gap: var(--spacing-1x);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.25rem;
  min-height: var(--size-3x);
  padding: 0 var(--spacing-1halfx);
  position: relative;
}

.c1:hover {
  background-color: transparent;
  color: #000000;
}

.c1:hover::before {
  background-color: rgb(0 0 0 / 0.05);
  border-radius: var(--border-radius);
  content: " ";
  display: block;
  height: calc(100% - 0.375rem);
  left: 0.1875rem;
  min-height: 1.5rem;
  position: absolute;
  top: 0.1875rem;
  width: calc(100% - 0.375rem);
}

.c1:focus {
  z-index: 1;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1[aria-disabled='true'],
.c1[aria-disabled='true']:hover {
  color: #B7BBC2;
  pointer-events: none;
}

.c1 span::after {
  content: attr(data-text);
  display: block;
  font-weight: var(--font-semi-bold);
  height: 0;
  overflow: hidden;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  visibility: hidden;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #E0F0F9;
  border: 1px solid #006296;
  border-radius: var(--border-radius);
  color: #003A5A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: var(--font-family);
  font-size: 1rem;
  font-weight: var(--font-semi-bold);
  gap: var(--spacing-1x);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  line-height: 1.25rem;
  min-height: var(--size-3x);
  padding: 0 var(--spacing-1halfx);
  position: relative;
}

.c2:hover {
  background-color: #C2E2F4;
  color: #003A5A;
}

.c2:hover::before {
  background-color: rgb(0 0 0 / 0.05);
  border-radius: var(--border-radius);
  content: none;
  display: block;
  height: calc(100% - 0.375rem);
  left: 0.1875rem;
  min-height: 1.5rem;
  position: absolute;
  top: 0.1875rem;
  width: calc(100% - 0.375rem);
}

.c2:focus {
  z-index: 1;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2[aria-disabled='true'],
.c2[aria-disabled='true']:hover {
  color: #B7BBC2;
  pointer-events: none;
}

.c2 span::after {
  content: attr(data-text);
  display: block;
  font-weight: var(--font-semi-bold);
  height: 0;
  overflow: hidden;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  visibility: hidden;
}

<div
  aria-label="Test4"
  class="c0"
  role="group"
>
  <button
    aria-pressed="false"
    class="c1"
    data-testid="test-toggle-button-0"
    type="button"
    value="option1"
  >
    <span
      data-text="Option 1"
    >
      Option 1
    </span>
  </button>
  <button
    aria-pressed="true"
    class="c2"
    data-testid="test-toggle-button-1"
    type="button"
    value="option2"
  >
    <span
      data-text="Option 2"
    >
      Option 2
    </span>
  </button>
  <button
    aria-disabled="true"
    aria-pressed="false"
    class="c1"
    data-testid="test-toggle-button-2"
    type="button"
    value="option3"
  >
    <span
      data-text="Option 3"
    >
      Option 3
    </span>
  </button>
  <button
    aria-pressed="false"
    class="c1"
    data-testid="test-toggle-button-3"
    type="button"
    value="option4"
  >
    <span
      data-text="Option 4"
    >
      Option 4
    </span>
  </button>
</div>
`;
